pyrevitlib/pyrevit/output/__init__.py

```diff
@@ -573,56 +654,127 @@ def print_table(self, table_data, columns=None, formats=None,
             title="Example Table",
             columns=["Row Name", "Column 1", "Column 2", "Percentage"],
             formats=['', '', '', '{}%'],
-            last_line_style='color:red;'
+            last_line_style='color:red;',
+            col_head_align_styles = ["left", "left", "center", "right"],
+            col_data_align_styles = ["left", "left", "center", "right"],
+            column_widths = ["100px", "100px", "500px", "100px"],
+            column_vertical_border_style = "border:black solid 1px",
+            table_width_style='width:100%', 
+            repeat_head_as_foot=True,
+            row_striping=False
+
             )
+            Returns:
+                Directly prints:
+                    print_md of the title (str): 
+                    print_html of the generated HTML table with formatting.
             ```
         """
-        if not columns:
-            columns = []
-        if not formats:
-            formats = []
 
-        if last_line_style:
-            self.add_style('tr:last-child {{ {style} }}'
-                           .format(style=last_line_style))
+        # Get user formatting options from kwargs
+        column_head_align_styles     = kwargs.get("column_head_align_styles", None)
+        column_data_align_styles     = kwargs.get("column_data_align_styles", None)
+        column_widths                = kwargs.get("column_widths", None)
+        column_vertical_border_style = kwargs.get("column_vertical_border_style", None)
+        table_width_style            = kwargs.get("table_width_style", None)
+        repeat_head_as_foot          = kwargs.get("repeat_head_as_foot", False)
+        row_striping                 = kwargs.get("row_striping", True)
+
+
+        # Set a unique ID for each table
+        # This is used in HTML tags to define CSS classes for formatting per table
+        table_uid = random.randint(10000, 99999) 
```

Comment:
better use a counter, there is a small chance you hit a bug 
set the variable somewhere higher:
`self._table_counter = 0`
then 
```
self._table_counter += 1
table_uid = self._table_counter
```